.travis.yml

```diff
@@ -12,33 +11,34 @@ arch:
 services:
   - docker  
 
+# Save up the cargo build cache
+cache: cargo  
+
 before_install:
 - rustup component add rustfmt
 
 # cargo test
 rust:
   - stable
   - beta
-  - nightly
 jobs:
-  allow_failures:
-    - rust: nightly
   fast_finish: true
 
-# Cargo fmt
-script:
-- cargo build
-- cargo test
-- cargo fmt -- --check
+# Build Test Fmt
+script: ./scripts/build.sh
+
+# Running deb build scripts
+script: ./rustscan-debbuilder/run.sh
 
 # Add in the github release details here
-# deploy:
-#   provider: releases
-#   api_key: "GITHUB OAUTH TOKEN"
-#   file: "FILE TO UPLOAD"
-#   skip_cleanup: true
-#   on:
-#     tags: true
+deploy:
+  provider: releases
+  api_key: $GH_0AUTH_TOKEN
+  file: target/debug/rustscan
+  file: debs/*.deb
+  skip_cleanup: true
+  on:
+    tags: true
```

Comment:
Does this deploy on push to master? We had a discussion and we'd like this workflow:
* push to master
* new github release automatically created (y.y.x where x is +1)
:sparkle:

Reply:
> new github release automatically created (y.y.x where x is +1)

The only downside with this approach is that we might create minor versions that are backward incompatible, which according to semantic versioning shouldn't happen. 😄 People expect that version `y.y.5` won't break after they upgrade from `y.y.4` for example.

Reply:
What about releasing on minor? `y.4.y`? 
**Advantages**
* BlackArch & maybe some other community packages build from master branch, by using master as our release we can better provide a guarantee to those using master for their packages
* We have pretty nice CI, a lot of unit tests and soon great code coverage. Not to mention every bit of code has to be reviewed, and we run the code before creating a PR to check for ourselves. At the moment we do not have any specific tests only for releases
* It's pretty cool to be able to say "master branch" == (docker, cargo, AUR, .deb releases) and may prevent users from building master themselves if they know our distributions are just as up to date as mater.

**Disadvantages**
* It can get quite messy if we release to Master a lot but do not release a major version, as in the minor version may reach some high numbers.
* We would release on minor pull requests such as spelling mistakes. To counter this, we'd have to merge pull requests together and create a new pull request (or update an existing one)

Reply:
The disadvantages you listed will become more apparent as the project grows. Right now we have these pain points because the project is growing fast, but as it stabilizes we will want to control our releases more tightly in order to not break peoples expectations and force then to constantly update their binaries.

Reply:
Fair enough! We'll use tagged releases :)